// Copyright (c) 2016, Google, Inc.  All rights reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto2";

package domain_policy;

// This is used to request a signed cert from the domain service.
// attestation is the marshaled attestation.
// program_key is the der encoded program public key.
// cert chain is any supporting certificates.
message DomainCertRequest {
  optional bytes attestation = 1;
  optional string key_type = 2;
  optional bytes subject_public_key = 3;
  // Consider making this a repeated when it's used.
  optional bytes cert_chain = 4;
}

// This is the response from the domain service.
// signed_cert is the signed program cert.
// cert_chain is the supporting certificate chain.
message DomainCertResponse {
  required int32 error = 1;
  optional bytes signed_cert = 2;
  repeated bytes cert_chain = 3;
}

